syntax = "proto3";

import public "clips_facts.proto";
import public "file.proto";

package buaa.dss;

//CLIPS Reasoning Service
service Reasoning{
  //@stub -param: facts -return: resoning results
  rpc GetReasoningResult(buaa.dss.placeholder) returns (buaa.facts.result){}

  //@stub
  rpc StartReasoning(buaa.facts.sim_start) returns (buaa.dss.placeholder){}

  //@stub -param:  -return:rule file list
  rpc GetRuleFile(buaa.dss.placeholder) returns (stream buaa.file.rule_file_info){}

  //@stub -param: the rule files that wanna loaded  -return:null
  rpc LoadRuleFile(stream buaa.file.rule_file_chk) returns (buaa.dss.placeholder){}

  //@stub Get the list of slots according to the chosen rule files
  rpc GetFactSlots(buaa.dss.placeholder) returns (stream buaa.file.fact_list){}

  //@stub Set the list of slots that selected by user
  rpc SetFactSlots(stream buaa.file.fact_list) returns(buaa.dss.placeholder){}

  //@stub -param: the rule files that wanna loaded  -return:optional rule list
  rpc GetFactFile(buaa.dss.placeholder) returns (stream buaa.file.fact_file_info){}

  //@stub -param: optional rule list  -return:null
  rpc LoadFactFile(stream buaa.file.fact_file_chk) returns (buaa.dss.placeholder){}

  //@stub Get the Slot list that selected before
  rpc GetSelectedSlots(buaa.dss.placeholder) returns (stream buaa.file.fact_list){}

  //@stub Get the optional variables from the facts file
  rpc GetVariableList(buaa.dss.placeholder) returns (stream buaa.file.fact_var){}

  //@stub Set the <slot:variable> kv mapping
  rpc SetMappingList(stream buaa.file.mapper) returns (buaa.dss.placeholder){}

}


message placeholder{}